src/main/scala/com/ing/wbaa/rokku/sts/service/UserTokenDbService.scala

```diff
@@ -101,11 +107,14 @@ trait UserTokenDbService extends LazyLogging with TokenGeneration {
    * Adds a user to the DB with aws credentials generated for it.
    * In case the user already exists, it returns the already existing credentials.
    */
-  private[this] def getOrGenerateAwsCredential(userName: UserName): Future[AwsCredential] =
+  private[this] def getOrGenerateAwsCredential(userName: UserName): Future[(AwsCredential, Boolean)] =
```

Comment:
maybe put the status in `AwsCredential` ?
```scala
  val status: Boolean
```
???
```suggestion
  private[this] def getOrGenerateAwsCredential(userName: UserName): Future[(AwsCredential)] =
```